frontend/ui/widget/filechooser.lua

```diff
@@ -394,18 +394,30 @@ function FileChooser:showSetProviderButtons(file, filemanager_instance, reader_u
 
     local buttons = {}
     local radio_buttons = {}
+    local filetype_provider = G_reader_settings:readSetting("provider") or {}
     local providers = DocumentRegistry:getProviders(file)
-
-    for ___, provider in ipairs(providers) do
-        -- we have no need for extension, mimetype, weights, etc. here
-        provider = provider.provider
+    if providers ~= nil then
+        for ___, provider in ipairs(providers) do
+            -- we have no need for extension, mimetype, weights, etc. here
+            provider = provider.provider
+            table.insert(radio_buttons, {
+                {
+                    -- @translators %1 is the provider name, such as Cool Reader Engine or MuPDF.
+                    text = T(_("%1 ~Unsupported"), provider.provider_name),
+                    checked = DocumentRegistry:getProvider(file) == provider,
+                    provider = provider,
+                },
+            })
+        end
+    else
+        local provider = DocumentRegistry:getProvider(file)
         table.insert(radio_buttons, {
             {
-                text = provider.provider_name,
-                checked = DocumentRegistry:getProvider(file) == provider,
+                text = provider.provider_name .. _(" ~Unsupported"),
```

Comment:
It should still read something like:
```suggestion
                T(_("%1 ~Unsupported"), provider.provider_name),
```

Reply:
sorry i cahnged the wrong one last time. Fixed